gamemodes/terrortown/entities/weapons/weapon_ttt2_sidekickdeagle.lua

```diff
@@ -136,4 +191,26 @@ if CLIENT then
 		local text = LANG.GetParamTranslation("ttt2_siki_shot", {name = target:GetName()})
 		MSTACK:AddMessage(text)
 	end)
+
+	net.Receive("tttSidekickRefillCDReduced", function()
+		if not timer.Exists("ttt2_sidekick_deagle_refill_timer") then return end
+
+		local timeLeft = timer.TimeLeft("ttt2_sidekick_deagle_refill_timer")
+		local newTime = math.max(timeLeft - GetConVar("ttt2_siki_deagle_refill_cd_per_kill"):GetInt(), 0.1)
+		local wep = LocalPlayer():GetWeapon("weapon_ttt2_sidekickdeagle")
+		timer.Adjust("ttt2_sidekick_deagle_refill_timer", newTime, 1, function() SidekickDeagleRefilled(wep) end)
```

Comment:
If LocalPlayer doesn‘t have the wep, this fn should return before adjusting the timer. Rare and tricky situation, but through networking delay possible